x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/timeline/timeline.tsx

```diff
@@ -0,0 +1,209 @@
+  if (!isFinite(ms)) {
+    return 100;
+  }
+  if (totalMs === 0) {
+    return 100;
+  }
+  return toPercent(ms, totalMs);
+};
+
+/**
+ * Each phase, if active, should have a minimum width it occupies. The higher this
+ * base amount, the smaller the variance in phase size in the timeline. This functions
+ * as a min-width constraint.
+ */
+const SCORE_BUFFER_AMOUNT = 50;
+
+const i18nTexts = {
+  hotPhase: i18n.translate('xpack.indexLifecycleMgmt.timeline.hotPhaseSectionTitle', {
+    defaultMessage: 'Hot phase',
+  }),
+  warmPhase: i18n.translate('xpack.indexLifecycleMgmt.timeline.warmPhaseSectionTitle', {
+    defaultMessage: 'Warm phase',
+  }),
+  coldPhase: i18n.translate('xpack.indexLifecycleMgmt.timeline.coldPhaseSectionTitle', {
+    defaultMessage: 'Cold phase',
+  }),
+  deleteIcon: {
+    toolTipContent: i18n.translate('xpack.indexLifecycleMgmt.timeline.delteIconToolTipContent', {
+      defaultMessage: 'Data will be deleted at the end of this policy',
+    }),
+  },
+};
+
+const calculateWidths = (inputs: PhaseAgeInMilliseconds) => {
+  const hotScore = msTimeToOverallPercent(inputs.phases.hot, inputs.total) + SCORE_BUFFER_AMOUNT;
+  const warmScore =
+    inputs.phases.warm != null
+      ? msTimeToOverallPercent(inputs.phases.warm, inputs.total) + SCORE_BUFFER_AMOUNT
+      : 0;
+  const coldScore =
+    inputs.phases.cold != null
+      ? msTimeToOverallPercent(inputs.phases.cold, inputs.total) + SCORE_BUFFER_AMOUNT
+      : 0;
+
+  const totalScore = hotScore + warmScore + coldScore;
+  return {
+    hot: `${toPercent(hotScore, totalScore)}%`,
+    warm: `${toPercent(warmScore, totalScore)}%`,
+    cold: `${toPercent(coldScore, totalScore)}%`,
+  };
+};
+
+const TimelinePhaseText: FunctionComponent<{
+  phaseName: string;
+  durationInPhase?: React.ReactNode | string;
+}> = ({ phaseName, durationInPhase }) => (
+  <EuiFlexGroup justifyContent="spaceBetween" gutterSize="none">
+    <EuiFlexItem>
+      <EuiText size="s">
+        <strong>{phaseName}</strong>
+      </EuiText>
+    </EuiFlexItem>
+    <EuiFlexItem grow={false}>
+      {typeof durationInPhase === 'string' ? (
+        <EuiText size="s">{durationInPhase}</EuiText>
+      ) : (
+        durationInPhase
+      )}
+    </EuiFlexItem>
+  </EuiFlexGroup>
+);
+
+export const Timeline: FunctionComponent = () => {
+  const [formData] = useFormData<FormInternal>();
+
+  const phaseTimingInMs = useMemo(() => {
+    return calculateRelativeTimingMs(formData);
+  }, [formData]);
+
+  const humanReadableTimings = useMemo(() => normalizeTimingsToHumanReadable(phaseTimingInMs), [
+    phaseTimingInMs,
+  ]);
+
+  const widths = calculateWidths(phaseTimingInMs);
+
+  const getDurationInPhaseContent = (phase: PhasesExceptDelete): string | React.ReactNode =>
+    phaseTimingInMs.phases[phase] === Infinity ? <InfinityIcon /> : humanReadableTimings[phase];
+
+  return (
+    <EuiFlexGroup gutterSize="s" direction="column" responsive={false}>
```

Comment:
I don't think the outer flex group is needed, it looks like it lays out the title and the timeline beyond, wdyt?

Reply:
If I understand correctly, are you saying we should replace the outer flex group with something like a `div` or `<>`? At the moment I am using EUI's flex group primarily for the `responsive=false` and gutter size. I guess we could handle this ourselves but it seemed like the easiest way of communicating layout. Let me know if I missed your point!

Reply:
Yes, I thought a `<>` could work to wrap the title and the timeline and a `EuiSpacer` for some white space between them. I was kinda expecting the elements inside flex group to be in a horizontal layout, hence my comment. But nothing blocking here.